python/Ganga/test/GPI/FileTests/TestMassStorageGetPut.py

```diff
@@ -0,0 +1,141 @@
+    _managed_files = []
+
+    # Num of sj in tests
+    sj_len = 3
+
+    # Where on local storage we want to have our 'MassStorage solution'
+    outputFilePath = '/tmp/TestMassStorageGetPut'
+
+    # This sets up a MassStorageConfiguration which works by placing a file on local storage somewhere we can test using standard tools
+    MassStorageTestConfig = {'defaultProtocol': 'file://',
+                             'fileExtensions': [''],
+                             'uploadOptions': {'path': outputFilePath, 'cp_cmd': 'cp', 'ls_cmd': 'ls', 'mkdir_cmd': 'mkdir -p'},
+                             'backendPostprocess': {'LSF': 'client', 'LCG': 'client', 'ARC': 'client', 'Dirac': 'client',
+                                                    'PBS': 'client', 'Interactive': 'client', 'Local': 'client', 'CREAM': 'client'}}
+
+    def setUp(self):
+        """
+        Configure the MassStorageFile for the test
+        """
+        extra_opts=[('PollThread', 'autostart', 'False'),
+                    ('Local', 'remove_workdir', 'False'),
+                    ('TestingFramework', 'AutoCleanup', 'False'),
+                    ('Output', 'MassStorageFile', self.MassStorageTestConfig),
+                    ('Output', 'FailJobIfNoOutputMatched', 'True')]
+        super(TestMassStorageGetPut, self).setUp(extra_opts=extra_opts)
+
+    @classmethod
+    def cleanUp(cls):
+        """ Cleanup the current temp objects """
+
+        from Ganga.GPI import jobs
+        for j in jobs:
+            shutil.rmtree(j.backend.workdir, ignore_errors=True)
+            j.remove()
+
+        for file_ in cls._temp_files:
+            os.unlink(file_)
+        cls._temp_files = []
+
+        for file_ in cls._managed_files:
+            os.unlink(os.path.join(cls.outputFilePath, file_.namePattern))
+        cls._managed_files = []
+
+        shutil.rmtree(cls.outputFilePath, ignore_errors=True)
+
+    def test_a_test_put(self):
+        """Test that a job can be submitted with inputfiles in the input"""
+
+        from Ganga.GPI import MassStorageFile
+
+        _ext = '.root'
+        file_1 = generate_unique_temp_file(_ext)
+        file_2 = generate_unique_temp_file(_ext)
+        self._temp_files.append(file_1)
+        self._temp_files.append(file_2)
+        msf_1 = MassStorageFile(file_1)
+        msf_2 = MassStorageFile(file_2)
+        self._managed_files.append(msf_1)
+        self._managed_files.append(msf_2)
+        msf_1.put()
+        msf_2.put()
+
+        for file_ in [msf for msf in (msf_1, msf_2)]:
+            assert os.path.isfile(os.path.join(self.outputFilePath, file_.namePattern))
+            file_.localDir = ''
+            assert file_.localDir == ''
+
+    def test_b_test_get(self):
+        """Test that the files were made accessible to the WN area and collected as LocalFile objects in outputfiles"""
+
+        from Ganga.GPIDev.Base.Proxy import stripProxy
+        from Ganga.GPI import Job
+
+        tmpdir = '/tmp/testMassStorageGet'
```

Comment:
This should use a real temp-dir creation function like pytest's [`tmpdir`](http://doc.pytest.org/en/latest/tmpdir.html) to avoid clashes between concurrent runs of the testing.